org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserCodeCompiler.java

```diff
@@ -64,8 +67,16 @@ private CompilationTask getCompilationTask(DiagnosticCollector<JavaFileObject> d
       files.add(
           new JavaSourceFromString(projectFile.getClassName(), projectFile.getFileContents()));
     }
+    List<String> optionList = new ArrayList<String>();
+    optionList.add("-classpath");
+    // Include the neighborhood jar in the student code classpath so the student code can access neighborhood classes.
+    try {
+      optionList.add(Paths.get(Objects.requireNonNull(getClass().getClassLoader().getResource("neighborhood-full.jar")).toURI()).toString());
```

Comment:
why do we need to add it to the classpath here as well as to the url class loader?

Reply:
My basic understanding: The classpath allows the compiler to make associations between classes whereas the url class loader pulls the classes into memory. I could definitely use clarification, however, as these are hypotheses based on what I was observing.

I was having a hard time find documentation explaining what I was observing. The basic observations, however, are these:
* Without using the classpath in the compiler, we get "package not found" errors. 
* Without loading the jar file at runtime, the JVM is able to see that the package exists, but is not able to find the classes. i.e. we get "class not found" errors.

Reply:
huh that's interesting. Everything I can find online says "class not found" occurs when the jar does not exist on the classpath.

Reply:
Hmm is that true as well for running it via Method.invoke as opposed to from the command line? 

Maybe there's a special setting we need to add to the compiler step so we don't need to use it at runtime?

Reply:
I wonder if this is the cause ([reference](https://www.geeksforgeeks.org/classnotfoundexception-vs-noclassdeffounderror-java/)):

> This exception also occurs when you have two class loaders and if a ClassLoader tries to access a class which is loaded by another classloader in Java.

Reply:
it may be that we have to load it both places--if we do, can we add a constant or helper method so we aren't referencing `"neighborhood-full.jar"` twice?